pkg/helpers/image/credentialprovider/config.go

```diff
@@ -147,6 +147,12 @@ func ReadSpecificDockerConfigJSONFile(filePath string) (cfg DockerConfig, err er
 	if contents, err = ioutil.ReadFile(filePath); err != nil {
 		return nil, err
 	}
+
+	ok := json.Valid(contents)
```

Comment:
Why the extra valid call, if `json.Unmarshal` invoked in `readDockerConfigJSONFileFromBytes` already does it? Why not wrapping the error returned from that method with information about file name instead?

Reply:
That's how I did it initially, but I wanted to avoid modifying function `readDockerConfigJSONFileFromBytes` the function name says `read docker config JSON file from bytes`  so I thought that logically it's better to leave validation out of that function in `ReadSpecificDockerConfigJSONFile`.

Reply:
Right, that's why I'm suggesting to wrap the error returned from `readDockerConfigJSONFileFromBytes` with the file path, only. Iow:
```golang
cfg, err := readDockerConfigJSONFileFromBytes(contents)
if err !=  nil {
    return nil, fmt.Errorf("invalid config %q: %w", filePath, err)
}
return cfg, nil
```
wdyt?

Reply:
Unmarshaling could fail if there is either typo in the config or the JSON is not valid. I was checking explicitly for the Validity of the JSON and not the config typo. This will cover both. I guess it's enough to say config XYZ is wrong. At least you get a path, so you can investigate more.

Reply:
Sorry I was wrong. Unmarshal will return only if its not valid JSON.

Reply:
Agree with your approach, at least expose the file path to the config itself, so sysadmin can investigate the problem.